testflinger/__init__.py

```diff
@@ -64,8 +63,9 @@ def create_flask_app():
     # Otherwise load it from testflinger.conf in the testflinger dir
     config_file = os.environ.get("TESTFLINGER_CONFIG", "testflinger.conf")
     tf_app.config.from_pyfile(config_file, silent=True)
-    if not os.path.exists(tf_app.config["DATA_PATH"]):
-        os.makedirs(tf_app.config["DATA_PATH"])
+    if isinstance(tf_app.config["DATA_PATH"], str):
+        tf_app.config["DATA_PATH"] = Path(tf_app.config["DATA_PATH"])
```

Comment:
Since we first load from the Default config object, we know this will be either a path or a string. We can Path() it either way, the only issue would be if it were None or something like that, which shouldn't happen. So I think you could even just skip the conditional here.